src/mantle/mantle_wsi.c

```diff
@@ -235,13 +235,21 @@ static void recreateSwapchain(
     GetClientRect(hwnd, &clientRect);
     const VkExtent2D imageExtent = { clientRect.right, clientRect.bottom };
 
+    VkSurfaceCapabilitiesKHR surfaceCapabilities;
+    res = vki.vkGetPhysicalDeviceSurfaceCapabilitiesKHR(grDevice->physicalDevice, mSurface,
+                                                  &surfaceCapabilities);
+    if (res != VK_SUCCESS) {
+        LOGE("vkGetPhysicalDeviceSurfaceCapabilitiesKHR failed (%d)\n", res);
+        return;
+    }
+
     // Recreate swapchain
     const VkSwapchainCreateInfoKHR swapchainCreateInfo = {
         .sType = VK_STRUCTURE_TYPE_SWAPCHAIN_CREATE_INFO_KHR,
         .pNext = NULL,
         .flags = 0,
         .surface = mSurface,
-        .minImageCount = 3,
+        .minImageCount = MAX(3, surfaceCapabilities.minImageCount),
```

Comment:
```suggestion
        .minImageCount = MIN(3, surfaceCapabilities.maxImageCount),
```

Reply:
according to the spec maxImageCount may be 0, and then we would need additional logic to account for that, I think minImage is better

Reply:
Ok. Thanks for the heads-up!

Perhaps your original patch was right, then?

```suggestion
        .minImageCount = MAX(2, surfaceCapabilities.minImageCount),
```

Reply:
I don't think 2 vs 3 images is gonna make a big difference, but I might be wrong on that

Reply:
We can just do: `.minImageCount = surfaceCapabilities.minImageCount`

Reply:
sometimes the minImageCount can be 1 and idk if apps will like that